### PR TITLE
feat: Allow global level Response headers

### LINKED ISSUE #335: [Feature Request] Allow global level Response headers
**Current Behavior**

Ideally, I would want to add it in this PR: https://github.com/sansyrox/robyn/pull/331 but I think it is already long enough.
**Desired Behavior**

We have the ability to add global request headers in Robyn(see here: https://github.com/sansyrox/robyn/blob/c7443963859a39b1998e032bcb9e682fbe563dbc/src/server.rs#L229 )

Similarly, we would love to allow global level Response headers.

---
**Screenshots / Mockups**

**Alternatives**

---

### PR DESCRIPTION
**Pull request by ParthS007**
**Description**

This PR fixes #335 

- Add support for global level response headers
- @sansyrox  We can merge `add_response_header` & `add_request_header` and `remove_header` & `remove_response_header`?

### CODE REVIEW

**Comment by sansyrox** [comment]
Thanks for the PR @ParthS007 . Great work like always! 😄 

> We can merge add_response_header & add_request_header and remove_header & remove_response_header?

What do you mean by this?

**Comment by ParthS007** [comment]
> > We can merge add_response_header & add_request_header and remove_header & remove_response_header?
> 
> What do you mean by this?

Both the methods `add_response_header` and `add_request_header` have similar functionality, we can use only one method for adding and have a keyword argument specifying either to `insert` in `global_response` or `global_request` header. Same goes in removal of header.

Does this sounds clear?

**Comment by sansyrox** [comment]
> Does this sounds clear?

Ah, that makes sense. We can do that in the rust side of code. But for the python facing developer API, having separate functions allows the developers to make fewer errors. As in Robyn, we follow explicit functions for a lot of things.

Would you like to work on this(integrating the rust side) btw?

**Comment by ParthS007** [comment]
@sansyrox I can help with that, Can you please point me to the related code in source and I will start having a look.

Is there anything else to add in this PR?

**Comment by sansyrox** [comment]
@ParthS007 , I was talking about this function(https://github.com/sansyrox/robyn/pull/410/files#diff-48d27698cc708c7efe770fd897e4885efaa0a15cae30d54936068603ba03bbd9R241) and its equivalent.

But I have been thinking about this. Do you think this optimization is necessary for a two-line function? Would a separation make more sense?

**Comment by sansyrox** [comment]
> Is there anything else to add in this PR?

@ParthS007 ,

I think we can merge this 😄

**Review by sansyrox — APPROVED** [review]
LGTM! Great work! 😄 ✨

**Comment by sansyrox** [comment]
@ParthS007 , a few tests are failing. Can you please rebase to the latest branch?

**Review by sansyrox — CHANGES_REQUESTED** [review]
Hey @ParthS007 ,

With the change in definition of `spawn_process`, we will also have to change the args here(https://github.com/sansyrox/robyn/blob/09e24ef6ee1b64e89a277e17b72303bb86a4f9bc/robyn/processpool.py#L72) and here(https://github.com/sansyrox/robyn/blob/09e24ef6ee1b64e89a277e17b72303bb86a4f9bc/robyn/processpool.py#L88)

This is why the tests are failing

File: src/types.rs [hunk 1]
```diff
@@ -164,6 +164,7 @@ pub struct Response {
 
 #[pymethods]
 impl Response {
```
Comment (ParthS007) [inline src/types.rs:166]: @sansyrox  We need to make change here to check for headers and create the Response accordingly?

File: docs/features.md [hunk 2]
```diff
@@ -194,6 +202,14 @@ async def request_headers():
     }
 ```
 
+```python
+@app.get("/response_headers")
```
Comment (ParthS007) [inline docs/features.md:206]: We should have similar endpoint for `response_header` as for `request_header`?

File: src/types.rs (hunk 1 above)
Comment (sansyrox) [reply src/types.rs:166]: @ParthS007 , in the future yes. But would we need that for now too? 

I suppose you are suggesting for things like json content?

File: docs/features.md (hunk 2 above)
Comment (sansyrox) [reply docs/features.md:206]: @ParthS007 , we already have one here(https://github.com/sansyrox/robyn/pull/410/files#diff-aff15ced1fe37506f41f2e1d93a4d8312529259f7803f8634ffb5eac4de47823R196)

Is this the same one that you're suggesting?

**Review by sansyrox — APPROVED** [review]
LGTM. Can merge and then work on this

**Pull request merged**